### DISCUSSION TITLE
Use data and string as value of expression?

### DISCUSSION THREAD

**Discussion by yanaleksandrov** [alpinejs/alpine#2475] [1. Help] [answered]
I create a new directive:

```
Alpine.directive( 'log', ( el, { value, expression, modifiers }, { evaluateLater, effect, cleanup } ) => {
	let evaluate = evaluateLater( expression )
	effect( () => {
		evaluate( content => {
			console.log(content)
		} )
	} )
} )
```

When I use second element, I'm catching an error: "Alpine Expression Error: Unexpected identifier".
```
<div x-data="{ message: 'Hello World!' }">
    <div x-log="message"></div>
</div>
```
```
<div x-data>
    <div x-log="Hello World!"></div>
</div>
```

As an expression, I want to use and string and data. Is it possible?

**Comment by SimoTod** [comment, marked as answer]
The second example should be `x-log="'Hello World'"`. The content of the attribute should be a valid js statement, if you want to use a string, you need quotes otherwise it thinks that `Hello` and `World!`are variables

**Reply by yanaleksandrov** [reply]
Thanks, it works!
